tests/test_iam_rules.py

```diff
@@ -0,0 +1,73 @@
+"""Tests  for the ARNRules."""
+
+import json
+
+from policyuniverse.policy import Policy
+
+from graph_altimeter.scan.iam import ARNRule
+
+
+def test_match_rule_global_resource():
+    """Tests a rule a matches a global aws resource, that is not tied to a
+    region."""
+    document = """
+    {
+        "Statement": [
+            {
+                "Action": [
+                    "s3:GetObject",
+                    "s3:PutObject"
+                ],
+                "Effect": "Allow",
+                "Resource": [
+                    "arn:aws:s3:::example-bucket"
+                ]
+            }
+        ],
+        "Version": "2012-10-17"
+        }
+    """
+    target_arn = "arn:aws:s3:eu-west-1:123456768124:bucket/example-bucket"
+    document = json.loads(document)
+    policy = Policy(document)
+    statement = policy.statements[0]
+    resource = statement.resources.pop()
+    rule = ARNRule(
+                        statement.action_summary(),
+                        statement.actions_expanded,
+                        resource,
+                        statement.effect,
+            )
+    assert(rule.matches(target_arn) == {"Read", "Write"})
+
+
+def test_match_resource():
+    """Tests a rule a matches an aws resource"""
+    document = """
+        {
+            "Statement": [
+                {
+                    "Action": [
+                        "dynamodb:Scan"
+                    ],
+                    "Effect": "Allow",
+                    "Resource": [
+                        "arn:aws:dynamodb:*:*:table/table*"
+                    ]
+                }
+            ],
+            "Version": "2012-10-17"
+        }
+    """
+    target_arn = "arn:aws:dynamodb:eu-west-1:123456768124:table/table2"
+    document = json.loads(document)
+    policy = Policy(document)
+    statement = policy.statements[0]
+    resource = statement.resources.pop()
```

Comment:
Why do you use pop() here?

Reply:
Because the ``resources`` property is of type [set](https://docs.python.org/3/library/stdtypes.html#set), so using `pop` I get the only element from it.